### PR TITLE
fix(megatron): avoid checkpoint scheduler double step

### PR DESCRIPTION
**Pull request by waitong94**
## Summary

- Avoid manually stepping the optimizer parameter scheduler after a Megatron checkpoint load has already restored scheduler state.
- Add a CPU-only regression test for checkpoint-restored and manually-stepped scheduler initialization paths.

## Why

When `use_checkpoint_opt_param_scheduler` is enabled, Megatron checkpoint loading restores the scheduler state from the checkpoint. Calling `opt_param_scheduler.step(iteration * global_batch_size)` immediately after that adds the checkpoint progress a second time, which shifts the learning-rate schedule after resume. The effect is especially pronounced for non-uniform schedules such as cosine decay, where double-counting can move the resumed run to a materially different learning rate.

The manual step is still needed when the checkpoint did not restore scheduler state, so the guard only skips it for nonzero checkpoint iterations with checkpoint scheduler restoration enabled.

## Validation

- `python -m pytest tests/fast/backends/megatron_utils/test_model_initialize.py -q`
- `python -m py_compile miles/backends/megatron_utils/model.py tests/fast/backends/megatron_utils/test_model_initialize.py`
- `pre-commit run --files miles/backends/megatron_utils/model.py tests/fast/backends/megatron_utils/test_model_initialize.py`

The new restored-checkpoint scheduler test fails on the previous implementation because `step(increment=800)` is called after loading iteration 100 with global batch size 8, and passes with this guard.

### CODE REVIEW

File: tests/fast/backends/megatron_utils/test_model_initialize.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    },
+)
+_stub_module(
+    "miles.backends.megatron_utils.bridge_lora_helpers",
+    {
+        "_ensure_model_list": MagicMock(),
+        "_setup_lora_model_via_bridge": MagicMock(),
+    },
+)
+_stub_module("miles.backends.megatron_utils.model_provider", {"get_model_provider_func": MagicMock()})
```
[the remark this answers is not in the document]
Comment (waitong94) [reply tests/fast/backends/megatron_utils/test_model_initialize.py]: Addressed in 4996cedef2. The Megatron/SGLang module stubs now live in a module-scoped autouse pytest fixture that snapshots sys.modules before stubbing and restores it at teardown. The failing stage-a CPU job was consistent with this leak: tests/fast/utils/test_dumper_utils.py later observed the stubbed dumper environment and returned {}. The focused scheduler test passes locally after the change, and CI has restarted on the amended commit.

**Comment by Rockdu** [comment]
LGTM, thanks for fixing this important bug! @Zhichenzzz I think this PR is ready to merge

**Review by Zhichenzzz — APPROVED** [review]
